apps/aehttp/src/aehttp_helpers.erl

```diff
@@ -157,6 +159,29 @@ verify_key_in_state_tree(Key, StateTreeFun, Lookup, Entity) ->
         end
     end.
 
+verify_name(NameKey) ->
+    fun(_Req, State) ->
+        Name = maps:get(NameKey, State),
+        case aens:get_name_hash(Name) of
+            {ok, _} -> ok;
+            {error, Reason} ->
+                ReasonBin = atom_to_binary(Reason, utf8),
```

Comment:
As @lucafavatella pointed out in another PR it would be good to have specs that ensure we're getting atoms that will be converted to binary, so that we don't crash here.

Would you change spec of `aens:get_name_hash/1` from `{error, term()}` to `{error, atom()}`?